plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2408,9 +2408,164 @@ read_only::get_table_rows_result read_only::get_kv_table_rows_context( const rea
      walk_table_row_range( lb_itr, ub_itr, false );
    }
 
+   // Set result.next_key
+   set_kv_next_key(p.encode_type, index_type, result);
+
    return result;
 }
 
+template <typename UInt, typename T>
+UInt read_only::float_from_bytes(T& t, const std::string& next_key_bytes) const {
+    std::stringstream ss;
+    ss << std::hex << next_key_bytes;
+    UInt val;
+    ss >> val;
+    UInt mask = 0;
+    UInt signbit = (static_cast<UInt>(1) << (std::numeric_limits<UInt>::digits - 1));
+    if (!(val & signbit)) //flip mask if val is positive
+        mask = ~mask;
+    val ^=(mask | signbit);
+    return val;
+}
+
+template <typename T>
+void read_only::convert_from_bytes( T& t, const std::string& next_key_bytes) const {
+    if constexpr (std::is_floating_point_v<T>) {
+        if constexpr (sizeof(T) == 4) {
+            uint32_t val = float_from_bytes<uint32_t>(t, next_key_bytes);
+            std::memcpy(&t, &val, sizeof(T));
+        }else {
+            static_assert(sizeof(T) == 8, "Unknown floating point type");
+            uint64_t val = float_from_bytes<uint64_t>(t, next_key_bytes);
+            std::memcpy(&t, &val, sizeof(T));
+        }
+    }else if constexpr (std::is_integral_v<T>) {
+        std::stringstream ss;
+        ss << std::hex << next_key_bytes;
+        auto unsigned_val = static_cast<std::make_unsigned_t<T>>(t);
```

Comment:
"t" is not initialized.

Reply:
no, either it will get set, or we will get an exception

Reply:
@brianjohnson5972, do you mean "t" should be initialized? I didn't see any exception without initializing it.